src/signals-unix.c

```diff
@@ -700,12 +707,97 @@ void trigger_profile_peek(void)
         }
     }
     bt_size_cur = 0; // clear profile buffer
-    if (jl_profile_start_timer() < 0)
+    if (jl_profile_start_timer(0) < 0)
         jl_safe_printf("ERROR: Could not start profile timer\n");
     else
         profile_autostop_time = jl_hrtime() + (profile_peek_duration * 1e9);
 }
 
+void jl_profile_task_unix(size_t nthreads)
+{
+    if (jl_profile_is_buffer_full()) {
+        // Buffer full: Delete the timer
+        jl_profile_stop_timer();
+        return;
+    }
+
+    jl_task_t *t = jl_get_random_task();
+    assert(t == NULL || jl_is_task(t));
+    if (t == NULL) {
+        return;
+    }
+    int t_state = jl_atomic_load_relaxed(&t->_state);
+    if (t_state == JL_TASK_STATE_DONE) {
+        return;
+    }
+
+    jl_rec_backtrace(t);
+
+    // store threadid but add 1 as 0 is preserved to indicate end of block
+    bt_data_prof[bt_size_cur++].uintptr = 1; // dummy value for now... Is this ever used when outputting the profile?
+
+    // store task id (never null)
+    bt_data_prof[bt_size_cur++].jlvalue = (jl_value_t*)t;
+
+    // store cpu cycle clock. XXX(Diogo, Nick): why are we recording the cycleclock here?
+    bt_data_prof[bt_size_cur++].uintptr = cycleclock();
```

Comment:
Nathan guessed it's so some profile viewing tools can display the profile over time (rather than as a single aggregated flamegraph), although idk if any tools are using this. Still, seems trivial to keep the same as with the existing profiler

Reply:
TODO(Diogo): investigate why we need to measure the cycle-clock and whether we can just keep this line as it is for the task profiler.